src/collector/directory_entries.rs

```diff
@@ -0,0 +1,88 @@
+use std::fs::{self, DirEntry};
+use std::io::ErrorKind;
+use std::path::PathBuf;
+
+use crate::{report::ReportEntry, Collector, CrateInfo, Result};
+
+use super::CollectionError;
+
+/// List information about entries in a directory.
+///
+/// Limitations:
+/// * Is not recursive
+/// * Does not handle symbolic links
+/// * Only sizes of files are printed and not e.g. time of last modification
+///
+/// # Example
+///
+/// ```md
+/// #### File and dir
+///
+/// - file.txt, 14 bytes
+/// - some_dir
+///
+/// ```
+pub struct DirectoryEntries {
+    title: String,
+    path: PathBuf,
+}
+
+impl DirectoryEntries {
+    pub fn new(title: impl Into<String>, path: impl Into<PathBuf>) -> Self {
```

Comment:
I tend to use `AsRef` on interfaces, e.g.:
```rust
impl<'a> FileContent<'a> {
    pub fn new<P: AsRef<Path>>(title: &'a str, path: P) -> Self {
// …

impl SoftwareVersion {
    pub fn custom<S: AsRef<str>>(version: S) -> Self {
// …

impl EnvironmentVariables {
    pub fn list<S: AsRef<OsStr>>(list: &[S]) -> Self {
// …
```

I saw that you wrote you wanted the title to be owned, but I was wondering if there was still value in keeping the interfaces consistent (you could still `clone` inside the constructor, like we do for `SoftwareVersion`). That would be a useless double-conversion in case a `String` is already being constructed at the call site, but this is not really performance-critical code (it's still not great). It just feels a bit strange to deviate for this particular `Collector`.

Reply:
Big thanks for code review!

Yes, I deliberately side-stepped from being consistent in the interfaces (hence the wording "took some liberties") since the current interfaces looks unnecessarily limiting to me. But I might very well be missing something.

How would you make the following work with the current interfaces?

```rust
use bugreport::collector::FileContent;
fn new_with_title_from_local_variable_file_content() -> FileContent<'static> { // What lifetime?
    let local_variable = String::from("pretend this is dynamically constructed");
    FileContent::new(&local_variable, "/path") // .clone() does not help, right?
}
```

right now we get

```
error[E0515]: cannot return value referencing local variable `local_variable`
  --> tests/test_collector_directory_entries.rs:95:5
   |
95 |     FileContent::new(&local_variable, "/path") // .clone() does not help. right?
   |     ^^^^^^^^^^^^^^^^^---------------^^^^^^^^^^
   |     |                |
   |     |                `local_variable` is borrowed here
   |     returns a value referencing data owned by the current function
```

Reply:
Aha wait sorry I think I misinterpreted you. At first I thought you did not want the title owned even inside the struct? But now I think you are OK with that, but you would like the interfaces consistent? I fully support and understand that.

I pushed a fix for that now. Now the signatures are identical:

```
% cargo public-items | grep -e ::new          # and adjust manually a bit
pub fn bugreport::collector::DirectoryEntries::new<P: AsRef<Path>>(title: &'a str, path: P) -> Self
pub fn      bugreport::collector::FileContent::new<P: AsRef<Path>>(title: &'a str, path: P) -> Self
```